# Q&A Activity Week -08

In [2]:
import pandas as pd
import scipy.stats as stat
data=pd.read_csv('Advertising.csv')
data

Unnamed: 0     TV  Radio  Newspaper  Sales
0             1  230.1   37.8       69.2   22.1
1             2   44.5   39.3       45.1   10.4
2             3   17.2   45.9       69.3    9.3
3             4  151.5   41.3       58.5   18.5
4             5  180.8   10.8       58.4   12.9
..          ...    ...    ...        ...    ...
195         196   38.2    3.7       13.8    7.6
196         197   94.2    4.9        8.1    9.7
197         198  177.0    9.3        6.4   12.8
198         199  283.6   42.0       66.2   25.5
199         200  232.1    8.6        8.7   13.4

[200 rows x 5 columns]

# Do the following on the dataset

# 1. Explain the dataset

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  200 non-null    int64  
 1   TV          200 non-null    float64
 2   Radio       200 non-null    float64
 3   Newspaper   200 non-null    float64
 4   Sales       200 non-null    float64
dtypes: float64(4), int64(1)
memory usage: 7.9 KB


# 1.There are 200 rows and 5 columns within the given data set,all with non-null values.

# 2.Data set columns include TV, Radio, Newspaper, Sales and one unnamed column representing the product/serial number.

# 3.TV,Radio and Newspaper columns indicate the expenditure for the advertising the product.

# 2. Check the summary statistics and discuss the max, min, avg, median, and percentiles.

In [5]:
data.describe()

Unnamed: 0          TV       Radio   Newspaper       Sales
count  200.000000  200.000000  200.000000  200.000000  200.000000
mean   100.500000  147.042500   23.264000   30.554000   14.022500
std     57.879185   85.854236   14.846809   21.778621    5.217457
min      1.000000    0.700000    0.000000    0.300000    1.600000
25%     50.750000   74.375000    9.975000   12.750000   10.375000
50%    100.500000  149.750000   22.900000   25.750000   12.900000
75%    150.250000  218.825000   36.525000   45.100000   17.400000
max    200.000000  296.400000   49.600000  114.000000   27.000000

max shows the maximum value in each column, min shows the minimum value in each column, mean shows the average of each column, 50% shows the median of each column, 25% 50% and 75% gives the quaterly percentile of each column.

# 3. The manager wants to find out whether the same amount was spent for the three advertisements (TV , Radio and Newspaper). Comment on your findings.

Let's assume that:

Null Hypothesis, H0 : Same amount was spent for three advertisements (TV,Radio,Newspaper), ie µ1=µ2=µ3

Alternative Hypothesis, H1: Different amount was spent for three advertisements (TV,Radio,Newspaper), ie μ1,µ2 and µ3 are not equal

# significance level

Assuming the Confidence level for this test to be 95% & we'll set the level of Significance as alpha = 0.05.

If p value < 0.05, We reject the Null hypothesis H0

If p value > 0.05, We reject the Alternative hypothesis HA

In [8]:
pvalue=stat.f_oneway(data['TV'],data['Radio'],data['Newspaper'])
pvalue

F_onewayResult(statistic=358.8514595342597, pvalue=4.552931539744962e-103)

since p value = 4.552931539744962e-103 is less than alpha(level of significance) ie 4.552931539744962 x 10^-103 < 0.05

# Conclusion

We can reject the null hypothesis and accept the alternate hypothesis,i.e "Different amount was spent for three advertisements namely TV,Radio & Newspaper" ie μ1, µ2 and µ3 are not equal.

 